plugins/cc/ccalgo.ml

```diff
@@ -233,9 +233,9 @@ type node =
      term:term}
 
 module Constrhash = Hashtbl.Make
-  (struct type t = constr
-	  let equal = eq_constr_nounivs
-	  let hash = Constr.hash
+  (struct type t = Evd.evar_map * EConstr.constr
+          let equal (sigma,c) (_sigma,c') = EConstr.eq_constr_nounivs sigma c c'
+          let hash (sigma,c) = Constr.hash (EConstr.to_constr sigma c)
```

Comment:
If I understand correctly this will still call `Constr.hash` with open terms, right?

I guess that `EConstr` should really provide a hash function, but what equality to take here is a bit subtle indeed [do we really want to pay the cost of normalizing in the hash function? That could have a big performance impact] cc @ppedrot

Reply:
> If I understand correctly this will still call Constr.hash with open terms, right?

Indeed, so, something remains to be done eventually if the objective is to remove `Evar` from `kernel`.

Reply:
More precisely: I have to backtrack on my claim that this PR (virtually) gets rid of `abort_on_undefined_evars:false` in `cc` after #6454 (or alternatively of a call to `unsafe_to_constr` on `nf_evars`, which amounts to the same). Hash on `EConstr` is needed to get rid of `abort_on_undefined_evars:false`.